### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by amirahmedimtiaz**

### CODE REVIEW

File: material-switch/src/switchModel.ets [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export namespace switchModel {
+  export class Model {
+    private switchId: number = 0
+    private switchSize: number = 16
```
Comment (archanaatr) [inline material-switch/src/switchModel.ets:19]: mark all dimentions as vp

File: material-switch/src/switchModel.ets [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+
+    getOnBackgroundColor(): string | Resource | Color {
+      return this.onBackgroundColor
+    }
+
+    setOnBackgroundColor(onBackgroundColor: string | Color | Resource): Model {
+      if (onBackgroundColor != null && onBackgroundColor!= undefined) {
+        this.onBackgroundColor = onBackgroundColor
+      } else {
+        this.onBackgroundColor = "rgba(55,136,62,1.0)"
```
Comment (archanaatr) [inline material-switch/src/switchModel.ets:44]: default values can be defines as constant in the top and used

File: material-switch/src/switchModel.ets [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+      }
+      else {
+        this.icon = $r('app.media.tick')
+      }
+      return this
+    }
+
+    reset(): Model {
+      this.switchId = 0
+      this.switchSize = 16
```
Comment (archanaatr) [inline material-switch/src/switchModel.ets:207]: all dimentions pls mark as vp

File: material-switch/src/switch.ets [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+
+@Component
+export struct Switch {
+  @State model: switchModel.Model = new switchModel.Model();
+  private onSelect: ((id: number, isOn: boolean) => void) = null
+
+  private onClick() {
+    if (this.onSelect != null) {
+      this.onSelect(this.model.getSwitchId(),!this.model.getIsOn());
```
Comment (archanaatr) [inline material-switch/src/switch.ets:26]: format code in all ets files

File: material-switch/src/switch.ets [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    }
+    if (this.model.getDisabled() == false) {
+      this.model.setShowAnimation(true)
+      animateTo({
+        duration: 100,
+        onFinish: () => {
+          if (this.model.getIsOn() == false) {
+            this.model.setSwitchSize(16)
```
Comment (archanaatr) [inline material-switch/src/switch.ets:34]: all dimentions in all the codes to be marked as vp

**Review by archanaatr — COMMENTED** [review]
please check the comments

File: .gitignore [hunk 6]
```diff
@@ -0,0 +1,5 @@
+/node_modules
```
Comment (archanaatr) [inline .gitignore:1]: remove .DS_Store file

File: README.en.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
+# Material_Switch
+
+This library is developed to provide customizable toggle switches  implemented using extended typescript.
+
+## Download & Install
+
+Install using npm
+
+```npm i @ohos-material-switch```
```
Comment (archanaatr) [inline README.en.md:9]: update library names as per the format shared

File: entry/src/main/config.json [hunk 8]
```diff
@@ -0,0 +1,68 @@
+{
```
Comment (archanaatr) [inline entry/src/main/config.json:1]: remove all these files
entry/src/main/.DS_Store
entry/src/main/config.json
.DS_Store

**Review by archanaatr — COMMENTED** [review]
pls check the comments.

File: material-switch/package.json [hunk 9]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "material-switch",
```
Comment (HMOS-Developer) [inline material-switch/package.json:2]: update name to - @ohos/material-switch

File: package.json [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+    "directoryLevel": "project"
+  },
+  "description": "example description",
+  "repository": {},
+  "license": "ISC",
+  "dependencies": {
+    "hypium": "^1.0.0",
+    "@ohos/hvigor": "1.0.6",
+    "@ohos/hvigor-ohos-plugin": "1.0.6",
+    "@ohos/material-switch": "file:./material-switch"
```
Comment (HMOS-Developer) [inline package.json:16]: remove this dependency from here

File: package.json [hunk 11]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "@ohos/material-switch",
```
Comment (HMOS-Developer) [inline package.json:2]: update name to - material-switch

File: entry/package.json [hunk 12]
```diff
@@ -0,0 +1,13 @@
...
+  "version": "1.0.0",
+  "ohos": {
+    "org": "huawei",
+    "buildTool": "hvigor",
+    "directoryLevel": "module"
+  },
+  "description": "example description",
+  "repository": {},
+  "license": "ISC",
+  "dependencies": {}
```
Comment (HMOS-Developer) [inline entry/package.json:12]: add library dependency here - "@ohos/material-switch": "file:./material-switch"

File: material-switch/src/switch.ets [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+
+
+@Component
+export struct Switch {
+  @State model: SwitchModel.Model = new SwitchModel.Model();
+  private onSelect: ((id: number, isOn: boolean) => void) = null
+
+  private onClick() {
+    if (this.onSelect != null) {
+      this.onSelect(this.model.getSwitchId(),!this.model.getIsOn());
```
Comment (archanaatr) [inline material-switch/src/switch.ets:26]: Code format to be done - Open each ets file and do Ctrl + Alt +  L

File: material-switch/src/switch.ets [hunk 14]
```diff
@@ -0,0 +1,90 @@
...
+      }
+      Circle()
+        .width(this.model.getShowAnimation() ? '28vp' : this.model.getSwitchSize())
+        .height(this.model.getShowAnimation() ? '28vp' : this.model.getSwitchSize())
+        .offset({
+          x: this.model.getShowAnimation() ?
+            (this.model.getIsOn() ? '20vp' : '2vp')
+                                           :
+            (this.model.getIsOn() ? '23vp' : '5vp'),
+          y: this.model.getShowAnimation() ? '2vp' : ((16 - (Number(this.model.getSwitchSize()
```
Comment (archanaatr) [inline material-switch/src/switch.ets:76]: if SwitchSize is 2vp, then substring will be '2v' and Number(2v) can cause abnormal exceptions, please consider such 
 cases and correct

File: material-switch/src/switchModel.ets [hunk 15]
```diff
@@ -0,0 +1,235 @@
...
+    private withIcon: boolean = false
+    private showAnimation: boolean = false
+    private isOn: boolean = false
+    private icon: Resource = $r('app.media.tick')
+    private disabled: boolean = false
+    private colorConstant = new colorConstants();
+    private onBackgroundColor: string | Color | Resource = this.colorConstant.defaultOnBackgroundColor
+    private offBackgroundColor: string | Color | Resource = this.colorConstant.defaultOffBackgroundColor
+
+
```
Comment (archanaatr) [inline material-switch/src/switchModel.ets:37]: remove multiple new empty lines

**Review by archanaatr — COMMENTED** [review]
please check the comments.

**Comment by amirahmedimtiaz** [comment]
switchSize variable was not supposed to be changed by user because code was written for specific measurements..So i have removed the  switchSize variable and modified and formatted the code

**Pull request merged**